fsspec/core.py

```diff
@@ -469,7 +471,11 @@ def open(
     return out[0]
 
 
-def open_local(url, mode="rb", **storage_options):
+def open_local(
+    url: str | list[str] | Path | list[Path],
+    mode: Optional[Literal["rb"]] = "rb",
```

Comment:
This annotation is wrong, I think, it implies mode can be ["rb", None] and nothing else. 
```suggestion
    mode: str = "rb",
```

The actual legal possibilities depend on the target filesystem, but include ["rb", "r", "rt", "w", "wb", "wt"]